src/NOTD.SC2Map/MapScript.galaxy

```diff
@@ -7791,77 +7791,85 @@ void gf_CharlieCompanyAI () {
+    }
+    while ((gv_gameOver == false)) {
+        UnitGroupLoopBegin(gv_easyCompanyTanks);
+        for ( ; !UnitGroupLoopDone() ; UnitGroupLoopStep() ) {
+            lv_escortDist = DistanceBetweenPoints(UnitGetPosition(UnitGroupLoopCurrent()), UnitGetPosition(UnitGroupClosestToPoint(lv_possibleEscorts, UnitGetPosition(UnitGroupLoopCurrent()))));
+            if ((UnitGetType(UnitGroupLoopCurrent()) == "SiegeBreaker") && (IntToFixed(UnitGroupCount(lv_siegeTanks, c_unitCountAlive)) < (IntToFixed(UnitGroupCount(gv_easyCompanyTanks, c_unitCountAlive)) * RandomFixed(0.33, 0.67))) && (lv_escortDist <= 3.0) && (lv_escortDist >= 1.35)) {
+                AISetUnitScriptControlled(UnitGroupLoopCurrent(), true);
+                UnitGroupAdd(lv_siegeTanks, UnitGroupLoopCurrent());
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("SiegeBreakerSiege", 0)), c_orderQueueReplace);
+            }
+            else if ((UnitGetType(UnitGroupLoopCurrent()) == "SiegeBreakerSieged") && ((lv_escortDist > 3.0) || (lv_escortDist < 1.35))) {
+                AISetUnitScriptControlled(UnitGroupLoopCurrent(), true);
+                UnitGroupRemove(lv_siegeTanks, UnitGroupLoopCurrent());
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("SiegeBreakerUnsiege", 0)), c_orderQueueReplace);
+                UnitGroupAdd(lv_unescorted, UnitGroupLoopCurrent());
             }
-
         }
-        lv_localIntA += 1;
+        UnitGroupLoopEnd();
+        Wait(2.1, c_timeGame);
+        UnitGroupLoopBegin(gv_easyCompany);
+        for ( ; !UnitGroupLoopDone() ; UnitGroupLoopStep() ) {
+            lv_escortDist = DistanceBetweenPoints(UnitGetPosition(UnitGroupLoopCurrent()), UnitGetPosition(UnitGroupClosestToPoint(lv_possibleEscorts, UnitGetPosition(UnitGroupLoopCurrent()))));
+            if ((lv_escortDist > 15.0)) {
+                if ((DataTableValueExists(false, ("esc_" + IntToString(UnitGetTag(UnitGroupLoopCurrent())))) == true)) {
+                    DataTableValueRemove(false, ("esc_" + IntToString(UnitGetTag(UnitGroupLoopCurrent()))));
+                }
+                AISetUnitScriptControlled(UnitGroupLoopCurrent(), true);
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("Sprint", 0)), c_orderQueueReplace);
+                UnitIssueOrder(UnitGroupLoopCurrent(), Order(AbilityCommand("Turbo", 0)), c_orderQueueReplace);
+                UnitIssueOrder(UnitGroupLoopCurrent(), OrderTargetingPoint(AbilityCommand("attack", 0), UnitGetPosition(UnitGroupClosestToPoint(lv_possibleEscorts, UnitGetPosition(UnitGroupLoopCurrent())))), c_orderQueueReplace);
+            }
+            else if ((((lv_escortDist <= 15.0) && (DataTableValueExists(false, ("esc_" + IntToString(UnitGetTag(UnitGroupLoopCurrent())))) == false)) || ((lv_escortDist > 7.0) && (DataTableValueExists(false, ("esc_" + IntToString(UnitGetTag(UnitGroupLoopCurrent())))) == true)))) {
```

Comment:
If the unit is within range but has no assigned escort, then assign an escort.
Or, if the unit has an assigned escort according to the local data table but is getting left behind, then the escort-ai might not be "active" for that unit, thus, reattempt assigning the chosen escort from the data table.